react/components/FilterSidebar.js

```diff
@@ -0,0 +1,157 @@
+import classNames from 'classnames'
+import PropTypes from 'prop-types'
+import {
+  map as mapRamda,
+  flatten,
+  filter,
+  pipe,
+} from 'ramda'
+import React, { Component, Fragment } from 'react'
+import { FormattedMessage } from 'react-intl'
+import { withRuntimeContext, Link } from 'render'
+import { Button } from 'vtex.styleguide'
+
+import { facetOptionShape } from '../constants/propTypes'
+import { mountOptions } from '../constants/SearchHelpers'
+import FilterIcon from '../images/FilterIcon'
+import AccordionFilterContainer from './AccordionFilterContainer'
+import Sidebar from './SideBar'
+
+class FilterSidebar extends Component {
+  static propTypes = {
+    map: PropTypes.string.isRequired,
+    rest: PropTypes.string.isRequired,
+    getLinkProps: PropTypes.func.isRequired,
+    runtime: PropTypes.object.isRequired,
+    filters: PropTypes.arrayOf(facetOptionShape).isRequired,
+    selectedFilters: PropTypes.arrayOf(facetOptionShape).isRequired,
+  }
+
+  state = {
+    openContent: false,
+    selectedFilters: pipe(
+      mapRamda(filter =>
+        mountOptions(filter.options, filter.type, this.props.map, this.props.rest),
+      ),
+      flatten,
+      filter(option => option.selected),
+      mapRamda(option => ({
+        ...option,
+        link: option.Link,
+      }))
+    )(this.props.filters || []),
+  }
+
+  handleFilterCheck = filter => {
+    if (!this.isOptionSelected(filter)) {
+      this.setState({
+        selectedFilters: this.state.selectedFilters.concat(filter),
+      })
+    } else {
+      this.setState({
+        selectedFilters: this.state
+          .selectedFilters
+          .filter(opt => opt.slug !== filter.slug),
+      })
+    }
+  }
+
+  handleClose = () => {
+    this.setState({
+      openContent: false,
+    })
+  }
+
+  handleOpen = () => {
+    this.setState({
+      openContent: true,
+    })
+  }
+
+  isOptionSelected = opt => (
+    !!this.state.selectedFilters.find(o => o.slug === opt.slug)
+  )
+
+  handleClearFilters = () => {
+    const { getLinkProps, runtime: { navigate } } = this.props
+
+    this.setState({ selectedFilters: [] })
+    const pagesArgs = getLinkProps([])
+
+    const options = {
+      page: pagesArgs.page,
+      params: pagesArgs.params,
+      query: pagesArgs.queryString,
+    }
+
+    navigate(options)
+  }
+
+  render() {
+    const { openContent, selectedFilters } = this.state
+    const {
+      filters,
+      map,
+      rest,
+      getLinkProps,
+    } = this.props
+
+    
+    const mountedFilters = mapRamda(x => 
+      ({ ...x, isSelected: false, link: x.Link }) , selectedFilters)
```

Comment:
move the `link` mapping to the `setState` on line 48 and also inline this map call directly to line 102.